fix: standardize Eventually() calls in auth tests

cam-garrison opened the pull request:
Fixes flaky test in CI by changing Expect()s to instead use returns.  Expect() was failing and panicking instantly, making the Eventually() clause in the test useless. Something we need to look out for in other tests as well.

bartoszmajsak commented:
@cam-garrison Could you expand on how these calls are different and what caused the panic before?

bartoszmajsak commented:
Now I understand the problem. Authz tests were not using `func (g Gomega)` passed to `Eventually` clause. Without it, `Eventually` block does not intercept errors, so it globally fails instead - therefore under certain conditions leading to failure as at the first run of it the requirements are not met yet.

Here's how it's done in routing controller tests and so should be in authz tests:

https://github.com/maistra/odh-platform/blob/df91330ec1d886847ffcbb120b03c5f2aaaf16ea/controllers/routingctrl/controller_test.go#L232-L262

A bit more details can be found in the [Gomega docs](https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode).

cam-garrison committed:
fix: standardize Eventually() calls

Since we were calling Eventually() without passing
(g Gomega, ctx ...) to the underlying func, our
tests in CI were flaky. By passing these, we fix
the flaky tests.

don't use Expect()s in flaky test

cam-garrison committed:
Merge branch 'main' into fix-flaky-test

bartoszmajsak committed:
Merge branch 'main' into fix-flaky-test
